src/App.js

```diff
@@ -14,11 +15,92 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.Component {
+  state = {
+    selectedGoods: [`Jam`],
+  }
+
+  clearSelection() {
+    this.setState({ selectedGoods: [] });
+  }
+
+  addGood(good) {
+    this.setState(state => (state.selectedGoods.push(good)));
+  }
+
+  removeGood(good) {
+    this.setState(
+      state => (
+        { selectedGoods: state.selectedGoods.filter(item => item !== good) }
+      ),
+    );
+  }
+
+  render() {
+    const { selectedGoods } = this.state;
 
+    return (
+      <div className="App box">
+        <div className="wrap">
+          <h1 className="row-with-items">
+            {!this.state.selectedGoods.length ? (
+              `No goods selected`
+            ) : (
+              `Selected products: ${selectedGoods}`
+            )}
+          </h1>
+          <button
+            className="button is-small is-danger"
+            type="button"
+            onClick={
+              () => this.clearSelection()
+            }
```

Comment:
```suggestion
            onClick={this.clearSelection}
```
если ты не передаешь в колбек параметры, можно писать так